packages/std/src/amp/messages.rs

```diff
@@ -262,10 +279,12 @@ impl AMPPkt {
         origin: impl Into<String>,
         previous_sender: impl Into<String>,
         messages: Vec<AMPMsg>,
+        previous_hops: Vec<Hop>,
```

Comment:
I think this is a pretty uncommon case to create a packet with hops directly other than in our Kernel. What do you think about just having this default to an empty vector and adding a `add_hop` method?